### PR TITLE
 Align with the block size of overlayBD

### PR DESCRIPTION
**Pull request by salvete**
**What this PR does / why we need it**:

The inconsistency in the default block sizes between OverlayBD and EROFS can lead to issues during segment mapping. This patch aligns with the block size of overlaybd for mapping (i.e., OverlayBD's 512-byte size) to overcome this inconsistency.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Please check the following list**:
- [ ]  Does the affected code have corresponding tests, e.g. unit test, E2E test?
- [ ]  Does this change require a documentation update?
- [ ]  Does this introduce breaking changes that would require an announcement or bumping the major version?
- [ ]  Do all new files have an appropriate license header?

### CODE REVIEW

**Comment by BigVan** [comment]
@hsiangkao PLTAL

File: src/overlaybd/tar/erofs/liberofs.cpp [hunk 1]
```diff
@@ -580,20 +580,27 @@ static int erofs_init_tar(struct erofs_tarfile *erofstar,
 static int erofs_write_map_file(photon::fs::IFile *fout, uint64_t blksz, FILE *fp)
 {
     uint64_t blkaddr, toff;
-    uint32_t nblocks;
+    uint32_t nblocks, zeroedlen;
+    char line[blksz];
```
Comment (hsiangkao) [inline src/overlaybd/tar/erofs/liberofs.cpp:584]: why blksz?

File: src/overlaybd/tar/erofs/liberofs.cpp [hunk 2]
```diff
@@ -580,25 +580,34 @@ static int erofs_init_tar(struct erofs_tarfile *erofstar,
...
     }
     rewind(fp);
-    while (fscanf(fp, "%" PRIx64" %x %" PRIx64 "\n", &blkaddr, &nblocks, &toff)
-           >= 3)
-    {
+
+    while (getline(&line, &len, fp) != -1) {
         LSMT::RemoteMapping lba;
+
+        cnt = sscanf(line, "%" PRIx64" %x %" PRIx64 "%u\n", &blkaddr, &nblocks, &toff, &zeroedlen);
```
Comment (hsiangkao) [inline src/overlaybd/tar/erofs/liberofs.cpp:597]: maybe `"%" PRIx64 "%x%" PRIx64 "%u"` is enough?

Comment (salvete) [reply src/overlaybd/tar/erofs/liberofs.cpp]: Done.

File: src/overlaybd/tar/erofs/liberofs.cpp [hunk 3]
```diff
@@ -580,25 +580,34 @@ static int erofs_init_tar(struct erofs_tarfile *erofstar,
...
     rewind(fp);
-    while (fscanf(fp, "%" PRIx64" %x %" PRIx64 "\n", &blkaddr, &nblocks, &toff)
-           >= 3)
-    {
+
+    while (getline(&line, &len, fp) != -1) {
         LSMT::RemoteMapping lba;
+
+        cnt = sscanf(line, "%" PRIx64" %x%" PRIx64 "%u", &blkaddr, &nblocks, &toff, &zeroedlen);
+
```
Comment (hsiangkao) [inline src/overlaybd/tar/erofs/liberofs.cpp:598]: I think you also need to bail out if `(cnt < 3)` and print some error messages.
if (cnt < 3)
     break;

Comment (salvete) [reply src/overlaybd/tar/erofs/liberofs.cpp:602]: Yes, that's better. Done.

**Review by hsiangkao — APPROVED** [review]

**Comment by BigVan** [comment]
thanks

File: src/overlaybd/tar/erofs/liberofs.cpp [hunk 4]
```diff
@@ -580,25 +580,38 @@ static int erofs_init_tar(struct erofs_tarfile *erofstar,
...
         lba.roffset = toff;
+        if (cnt > 3)
+            lba.count = round_up_blk(lba.count - zeroedlen);
+
         int nwrite = fout->ioctl(LSMT::IFileRW::RemoteData, lba);
         if ((unsigned) nwrite != lba.count) {
             LOG_ERRNO_RETURN(0, -1, "failed to write lba");
         }
     }
+    free(line);
```
Comment (hsiangkao) [inline src/overlaybd/tar/erofs/liberofs.cpp:614]: if (line)
    free(line);

**Pull request merged**